controller/token.go

```diff
@@ -212,6 +218,11 @@ func (c *TokenController) Exchange(ctx *app.ExchangeTokenContext) error {
 
 	if notApprovedRedirect != nil && token == nil {
 		// the code enters this block only if the user is not provisioned on OSO.
+		userProfileContext := profileCtx.Value(provider.UserProfileContextKey).(*provider.UserProfileContext)
```

Comment:
How about using `userProfileContext, ok := profileCtx.Value(provider.UserProfileContextKey).(*provider.UserProfileContext)` to avoid panic trigger?